backend/server/routers/courses.py

```diff
@@ -419,23 +419,28 @@ def weight_course(course: tuple, search_term: str, structure: dict,
     (code, title) = course
 
     if major_code is not None:
-        for key in structure['Major'].items():
-            if isinstance(key[1], dict):
-                if key[1].get(code) is not None:
-                    weight += 20
-                    break
+        for structKey in structure.keys():
+            if "Major" not in structKey:
+                continue
+            for key in structure[structKey].items():
+                if isinstance(key[1], list):
+                    if code in key[1]:
+                        weight += 10
```

Comment:
```suggestion
                        weight += 20
```
I'd rather this weighting stayed the same, maybe make it 15 if you feel like the results aren't what you want